src/main/java/screens/Window.java

```diff
@@ -214,46 +215,58 @@ private Menu addController() {
-                    btn.setOnAction(
-                            event2 -> {
-                                int radius = Integer.parseInt(textField2.getText());
-                                if (radius < 5 || radius > 500) {
-                                    Stage newStage = new Stage();
-                                    Group group = new Group();
-                                    Label label = new Label("Radius is out of bounds");
-                                    group.getChildren().add(label);
-                                    Scene scene = new Scene(group, 150, 100);
-                                    newStage.setScene(scene);
-                                    newStage.show();
-                                } else {
-                                    graphScene.drawGraph(Integer.parseInt(textField.getText()), Integer.parseInt(textField2.getText()));
-                                    graphScene.switchToInfo();
-                                    newstage.close();
+                    if (NodeGraph.getCurrentInstance() != null) {
+                        Stage newstage = new Stage();
+                        newstage.setTitle("Select the radius");
+                        GridPane box = new GridPane();
+                        TextField textField = new TextField();
+                        TextField textField2 = new TextField();
+                        Button btn = new Button("Submit");
+                        btn.setOnAction(
+                                event2 -> {
+                                    if (textField.getText().length() == 0 || textField.getText().contains("\\D")) {
+                                        errorPopup("Please enter a number as id.");
+                                    } else if (textField2.getText().length() == 0 || textField2.getText().contains("\\D")) {
+                                        errorPopup("Please enter a number as radius.");
+                                    } else {
+                                        int center = Integer.parseInt(textField.getText());
+                                        int radius = Integer.parseInt(textField2.getText());
+
+                                        if (center < 0 || center >= NodeGraph.getCurrentInstance().getSize()) {
+                                            errorPopup("Input center id is out of bounds, \nplease provide a different input id.");
+                                        } else if (radius < 5 || radius > 500) {
+                                            errorPopup("Input radius is out of bounds, \nplease provide a different radius.");
+                                        } else {
+                                            graphScene.drawGraph(Integer.parseInt(textField.getText()), Integer.parseInt(textField2.getText()));
+                                            graphScene.switchToInfo();
+                                            newstage.close();
```

Comment:
Okay, it does fall under the goal of making the codebase better, but I'll approve it for now, but it will need to be refactored some other time.